### ISSUE TITLE
Cancel endpoint on uploads

### ISSUE THREAD

**Issue by jkeifer** [PSU-CSAR/django-ebagis#40]
Create an endpoint `uploads/<id>/cancel` to stop uploads that will never be finished or are processing.

See PSU-CSAR/vb-bagis-h#13 and #8 for more information.

**Comment by jkeifer** [comment]
Just to add a little more documentation, this will return HTTP status codes as follows:
- 200 for a successful abort
- 400 if the abort was not successful (a task has already been completed or aborted)
- 404 if the upload was not found or a user did not have permission to abort said upload

**Comment by jkeifer** [comment]
Seems to be working, so I'm going to close this out.

**Comment by lbross** [comment]
Does this work for downloads too? If the user requests a download but changes their mind before the zip file is created, can they abort the task?

**Comment by jkeifer** [comment]
I remember we discussed this and determined that there was little to no value in offering this feature (at least that's where we arrived at the time). If I remember correctly, the reasoning was because we hope to implement download caching, so most downloads would be unlikely to trigger a task (as the most recent version would be cached daily), and even if they did, the zip could be cached for future use.

However, I'm open to revising this topic. The download tasks should support aborting using the same mechanism as the uploads, I just did not create the rest interface or model attributes for it.
